pdc/apps/package/serializers.py

```diff
@@ -221,3 +222,68 @@ class BuildImageRTTTestsSerializer(StrictSerializerMixin, serializers.ModelSeria
     class Meta:
         model = models.BuildImage
         fields = ('id', 'build_nvr', 'format', 'test_result')
+
+
+class RepositoryField(serializers.SlugRelatedField):
+    def __init__(self, **kwargs):
+        super(RepositoryField, self).__init__(slug_field='id', queryset=Repo.objects.all(), **kwargs)
+
+    def to_representation(self, value):
+        export_data = value.export()
+        export_data["id"] = value.id
+        return export_data
+
+
+class ReleasedFilesSerializer(StrictSerializerMixin, serializers.ModelSerializer):
+    # base on content.format, currently rpm.srpm_name, rpm.version, rpm.release
+    build = serializers.CharField(required=False, default=None)
+    # base on content.format, currently rpm.srpm_name
+    package = serializers.CharField(required=False, default=None)
+    # base on content.format, currently rpm.filename
+    file = serializers.CharField(required=False, default=None)
+    # base on content.format, pk
+    file_primary_key = serializers.IntegerField(allow_null=False)
+    repo = RepositoryField()
+    released_date = serializers.DateField(required=False)
+    release_date = serializers.DateField()
+    created_at = serializers.DateTimeField(required=False, read_only=True)
+    updated_at = serializers.DateTimeField(required=False, read_only=True)
+    zero_day_release = serializers.BooleanField(required=False, default=False)
+    obsolete = serializers.BooleanField(required=False, default=False)
+
+    class Meta:
+        model = models.ReleasedFiles
+        fields = ('id', 'build', 'package', 'file', 'file_primary_key', 'repo',
+                  'released_date', 'release_date', 'created_at', 'updated_at',
+                  'zero_day_release', 'obsolete')
+
+    def validate(self, data):
+        repo_format = data["repo"].content_format
```

Comment:
Test `'repo' in data` (and for other keys too) otherwise partial PATCH fails.

Reply:
You just can insert/change the repo.id here, the repo info can't change here.

Reply:
Let me be more clear:
```bash
> pdc_client -r released-files/1 -x PATCH -d '{"repo": 18396}'           
ERROR:pdc_client:503 {"detail":"The server encountered an internal error or misconfiguration and was unable to complete your request."}

> pdc_client -r released-files/1 -x PATCH -d '{"zero_day_release": "true"}'
ERROR:pdc_client:503 {"detail":"The server encountered an internal error or misconfiguration and was unable to complete your request."}
```

Reply:
Now it's more clear :), and very nice catch.
I have fixed it.